src/main/java/com/epam/xmlparsing/handler/DeviceHandler.java

```diff
@@ -0,0 +1,97 @@
+package com.epam.xmlparsing.handler;
+
+import com.epam.xmlparsing.entity.*;
+import com.epam.xmlparsing.parser.DeviceXmlTag;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+
+public class DeviceHandler extends DefaultHandler {
+    private final List<Device> devices = new ArrayList<>();
+    private Device currentDevice;
+    private DeviceXmlTag currentTag;
+    private final EnumSet<DeviceXmlTag> tagsWithText = EnumSet.range(DeviceXmlTag.ORIGIN, DeviceXmlTag.SCREENDIAMETER);
+
+    public List<Device> getDevices() {
+        return devices;
+    }
+
+    @Override
+    public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
+        String laptopDeviceTag = DeviceXmlTag.LAPTOP.getValue();
+        String smartphoneDeviceTag = DeviceXmlTag.SMARTPHONE.getValue();
+        if (laptopDeviceTag.equals(qName) || smartphoneDeviceTag.equals(qName)){
+            currentDevice = laptopDeviceTag.equals(qName) ? new Laptop() : new Smartphone();
+            defineAttributes(attributes);
+        }
```

Comment:
when you have else in if, you should keep closing curly brace and else within 1 line:
if(...){
    ....
} else {